### ISSUE TITLE
If-Range cannot be used correctly with etags

### ISSUE THREAD

**Issue by lolbinarycat** [hyperium/headers#194]
[a weak entity tag must not be used with If-Range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Range), but there is no easy way to check if an `ETag` is weak.

additionally, [`IfRange::etag`](https://docs.rs/headers/latest/headers/struct.IfRange.html#method.etag) is infallible, which means it is very easy to create an incorrect implementation.

**Comment by seanmonstar** [comment]
Indeed! [RFC 9110 Section 13.1.5](https://datatracker.ietf.org/doc/html/rfc9110#section-13.1.5):

> A client MUST NOT generate an If-Range header field containing an entity tag that is marked as weak.

Welp, that's an unfortunate oversight. We can include adjustments in the next breaking change (which are not as catastrophic, since the trait is separate from the impls.) Would do well to design what the changes are here.

**Comment by lolbinarycat** [comment]
At the very least we should add an `is_weak` method to `ETag` so this can be mitigated in the meantime.
